include/gauge_field.h

```diff
@@ -872,4 +893,12 @@ namespace quda {
 
 #define checkReconstruct(...) Reconstruct_(__func__, __FILE__, __LINE__, __VA_ARGS__)
 
+  /**
+   * @brief Generic gauge field shift
+   * @param[out] dst Gauge field to store output
+   * @param[in] srd Source from which we are shifting (extended field in case of MPI)
+   * @param[in] dx Host array of shifts to apply to the field
+   */
+  void gaugeShift(GaugeField &dst, const GaugeField &src, const array<int, 4> &dx);
```

Comment:
Should this be moved to e.g. `gauge_tools.h`?

Reply:
Yeah, that seems like a consistent place for it.